src/ucs/datastruct/array.h

```diff
@@ -210,6 +209,19 @@ BEGIN_C_DECLS
    UCS_ARRAY_IDENTIFIER(_name, _append)(_array)
 
 
+/*
+ * Add an element to the end of the array assuming it has enough space.
+ *
+ * @param _name     Array name
+ * @param _array    Array to add element to
+ */
+#define ucs_array_append_fixed(_name, _array) \
+    ({ \
+        ucs_assert_always(ucs_array_append(_name, _array) == UCS_OK); \
+        ucs_array_last(_array); \
+    })
```

Comment:
is this macro still needed?

Reply:
used in ucp_proto_perf_envelope_append()

Reply:
ok